Realm/RLMAnalytics.mm

```diff
@@ -101,22 +105,99 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
 
 // Get the version of OS X we're running on (even in the simulator this gives
 // the OS X version and not the simulated iOS version)
-static NSString *RLMOSVersion() {
-    std::array<int, 2> mib = {{CTL_KERN, KERN_OSRELEASE}};
+static NSString *RLMHostOSVersion() {
+    size_t size;
+    sysctlbyname("kern.osproductversion", NULL, &size, NULL, 0);
+    char *model = (char*)malloc(size);
+    sysctlbyname("kern.osproductversion", model, &size, NULL, 0);
+    NSString *deviceModel = [NSString stringWithCString:model encoding:NSUTF8StringEncoding];
+    free(model);
+    return deviceModel;
+}
+
+static NSString *RLMHostArch() {
+    std::array<int, 2> mib = {{CTL_HW, HW_MACHINE}};
```

Comment:
This does not get the host architecture, but we also don't report analytics in any scenario where the host and target architectures would be different so it doesn't really make sense to do in the first place.

Reply:
There used to be i386 simulators which would run on x86_64, but these days there's only x86_64 and arm64.

Reply:
mmmm I guess you are right, there is no case where the host architecture is different that the target architecture, I guess we could inform at least one of them. @tgoyne